### PR TITLE
refactor(builder): pass nuxt options to template as nuxtOptions

### LINKED ISSUE #5730: Cannot access templateVars.options in templates
### Version

[v2.7.0](https://github.com/nuxt.js/releases/tag/v2.7.0)

### Reproduction link

[https://codesandbox.io/s/946pw7j9k4](https://codesandbox.io/s/946pw7j9k4)

### Steps to reproduce

a) create a module that injects a plugin
b) have plugin use lodash syntax to fill in vars from templateVars
c) notice that even though templateVars.options exists, it's { } in the written template

The problems seems to be with https://github.com/nuxt/nuxt.js/blob/dev/packages/builder/src/builder.js#L515 which merges options with the template, even if options isn't passed. 

### What is expected ?

My assumption is that anything you see in templateVars should be accessible in a template!

### What is actually happening?

options prop is being reset

### Additional comments?

a) could maybe see if options is being passed to the template, then let it override it (which still is kind of crummy)
b) could move either options or templateVars.options into a new prop in templateVars so it's always accessible.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9215">#c9215</a>)</em></sub></div>

### LINKED ISSUE #5783: SyntaxError: /.nuxt/nuxt-icons.js: Unexpected token, expected "," (93:1387)
### Version

[2.7.2](https://github.com/nuxt.js/releases/tag/2.7.2)

### Reproduction link

[http://link.re](http://link.re)

### Steps to reproduce

npm run build

### What is expected ?

Normal building

### What is actually happening?

ERROR in ./.nuxt/nuxt-icons.js
Module build failed (from ./node_modules/babel-loader/lib/index.js):
SyntaxError: /.nuxt/nuxt-icons.js: Unexpected token, expected "," (93:1387)

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9249">#c9249</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Fixes: #5783
Related to #5730

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/builder/src/builder.js [hunk 1]
```diff
@@ -515,10 +515,8 @@ export default class Builder {
             templateFunction({
               ...templateVars,
               ...templateFile,
-              options: {
-                ...templateVars.options,
-                ...templateFile.options
-              }
+              options: templateFile.options || {},
+              nuxtOptions: templateVars.options
```
Comment (pi0) [inline packages/builder/src/builder.js]: Can we directly use `this.nuxt.options` here are remove it from templateVars? Could be useful for feature developments.

**Review by pi0 — DISMISSED** [review]
LGTM

**Review by galvez — APPROVED** [review]
Nice!

**Comment by pi0** [comment]
For docs, I suggest keeping this new (feature) as undocumented. In any stage, we may change the config interface in an incompatible way and this may break users. Maybe we can suggest always cherry-picking needed options into template `options` param when needed.

**Comment by clarkdo** [comment]
@pi0 Agree

**Pull request merged**